### ISSUE TITLE
Larger Batch Size Support

### ISSUE THREAD

**Issue by QAQdev** [LTH14/rcg#36]
Hello! Great work!

When I am reproducing LDM on my GPU A800 (80G), the configuration, i.e. batch size=4 (default) can not utilize full GPU memory.

Since I want to set larger batch size to make full use of GPU memory, I wonder if you could recommend a configuration of hyperparamters such as (`epochs`, `blr`, `weight_decay`) when batch size is set larger, for example, when batch size is set to 16 instead of 4, how to tune `epochs`, `blr`, `weight_decay` accordingly?

Thanks a lot!!

**Comment by LTH14** [comment]
Thanks for your interest! The learning rate should be related to your overall batch size, which is adjusted automatically here https://github.com/LTH14/rcg/blob/main/main_ldm.py#L149-L152. Epochs and weight decay can remain the same. In our case, we use 64 GPUs with batch size=4, which means an overall batch size=256. You could use 16 A100s with batch size 16, or 8 A100s with batch size 8 to achieve a similar performance.

**Comment by QAQdev** [comment]
> or 8 A100s with batch size **8** to achieve a similar performance.

I think it should be 32 here?

BTW, so by this explanation, I just need to make the global batch size = 256, and dont need to  tune epochs, blr, weight_decay?

**Comment by LTH14** [comment]
Yes -- global batch size=256 and other parameters should remain the same. In fact, even with a global batch size of 128-1024, the performance should be similar with the same set of parameters because we automatically adjust the learning rate according to batch size.

**Comment by QAQdev** [comment]
That's great, thanks for your in-time reply!

One more question, since I want to apply LDM on tinyimagenet, can I just follow this setting?
